### ISSUE TITLE
Package Rename Changelog Validation

### ISSUE THREAD

**Issue by kanthesha** [MetaMask/core#1795]
### Description:
In the process of migrating existing repos to the core mono repo, sometimes there'll be a change in the package name. This will impact the validation of changelog. In this ticket, we would like to explore the changes required in `metamask/auto-changelog`, so that it can accept the additional parameters to make sure, validations passes for old packages name with old tags as well as newly changed package name.

Example: Scenario
package name before moving into mono repo: `json-rpc-midddleware-stream`
package name after moving into mono repo: `@metamask/json-rpc-midddleware-stream`

An example of tags part of changelog.md from original/old package looks like below
```
[Unreleased]: https://github.com/MetaMask/json-rpc-middleware-stream/compare/v5.0.1...HEAD
[5.0.1]: https://github.com/MetaMask/json-rpc-middleware-stream/compare/v5.0.0...v5.0.1
[5.0.0]: https://github.com/MetaMask/json-rpc-middleware-stream/compare/v4.2.3...v5.0.0
```

An example of tags part of changelog.md from core mono repo
Line 1 - unreleased in core mono repo
Line 2 - migrated package released in core mono repo with version 6.0.0
rest of the lines are from the migrated repo
```
[Unreleased]: https://github.com/MetaMask/core/compare/@metamask/json-rpc-middleware-stream@6.0.0...HEAD
[6.0.0]: https://github.com/MetaMask/core/compare/json-rpc-middleware-stream@5.0.1...@metamask/json-rpc-middleware-stream@6.0.0
[5.0.1]: https://github.com/MetaMask/core/compare/json-rpc-middleware-stream@5.0.0...json-rpc-middleware-stream@5.0.1
[5.0.0]: https://github.com/MetaMask/core/compare/json-rpc-middleware-stream@4.2.3...json-rpc-middleware-stream@5.0.0
```

### DoD:
1. Changelog validation working fine for the old package name in the entries and also for the new package release entries
2. An option to configure Old Tag Prefix and Old Package Version

**Comment by kanthesha** [comment]
And right now we're only trying to implement for the one time package rename! But the ideal feature and implementation would be to work for package renaming even when it is multiple times, which requires an array of {version: string, tagPrefix: string}.

cc: @mcmire @desi

**Comment by mcmire** [comment]
This was completed via https://github.com/MetaMask/auto-changelog/pull/157 and is available in `@metamask/auto-changelog` 3.4.0.
